src/a11y-dialog.ts

```diff
@@ -204,7 +217,10 @@ export default class A11yDialog {
     // boundaries
     // See: https://github.com/KittyGiraudel/a11y-dialog/issues/712
     if (opener) this.show(event)
-    if (explicitCloser || implicitCloser) this.hide(event)
+    if (explicitCloser || implicitCloser) {
+      if (closeWatcher) closeWatcher.requestClose()
+      else this.hide(event)
+    }
```

Comment:
This is a breaking change, because we no longer can provide access to the “closer” (as in the element that was interacted with to close the dialog) via the close watcher. That’s because we can’t pass an event to the `.close(..)` or `.requestClose(..)` method which would be forwarded to the `onclose` callback. So this is a bit shitty.